### PR TITLE
Update firewalld resource to include more properties

### PR DESCRIPTION
**Pull request by dtingstrom**
I needed to retrieve the target of an active firewall zone and noticed that the target and some other properties were missing from the firewalld resource. So I added these 9 properties:
1. target
2. icmp_block_inversion
3. ports
4. protocols
5. masquerade
6. forward_ports
7. source_ports
8. icmp_blocks
9. rich_rules

I also created mock tests for all the new properties.

Signed-off-by: Daniel Tingstrom <dtingstrom@mitre.org>

### CODE REVIEW

File: lib/inspec/resources/firewalld.rb [hunk 1]
```diff
@@ -120,9 +129,72 @@ def parse_line(line)
...
       }
     end
 
+    def target_bound(query_zone)
+      # result: a target bound for the zone
+      # example: 'DROP'
+      firewalld_command("--permanent --zone=#{query_zone} --get-target").strip()
+    end
+
+    def icmp_block_inversion_bound(query_zone)
```
Comment (aaronlippold) [inline lib/inspec/resources/firewalld.rb:150]: Please add a `?` at the end of the function as this is a boolean per ruby standard conventions.

File: lib/inspec/resources/firewalld.rb [hunk 2]
```diff
@@ -120,9 +129,72 @@ def parse_line(line)
...
+      firewalld_command("--zone=#{query_zone} --list-ports").split(" ")
+    end
+
+    def protocols_bound(query_zone)
+      # result: a list of protocols added for a zone
+      # example: ['icmp', 'ipv4', 'igmp']
+      firewalld_command("--zone=#{query_zone} --list-protocols").split(" ")
+    end
+
+    def masquerade_bound(query_zone)
```
Comment (aaronlippold) [inline lib/inspec/resources/firewalld.rb:168]: Please add a ? at the end of the function as this is a boolean per ruby standard conventions.

**Review by aaronlippold — CHANGES_REQUESTED** [review]
Please add a `?` at the end of the function as this is a boolean per ruby standard conventions.

**Comment by wdower** [comment]
Reopening against inspec/inspec.

**Pull request closed without merging**